backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/ui/web/order/BOMExploderCommand.java

```diff
@@ -130,17 +139,59 @@ public BOMExploderCommand(@Nullable final BOMUse bomToUse,
 			final OrderLineCandidate lineCandidate = initialCandidate.toBuilder()
 					.productId(bomLineProductId)
 					.attributes(attributes)
-					.qty(bomLineQty)
+					.qty(Quantitys.create(bomLineQty, bomUomId))
 					.compensationGroupId(compensationGroupId)
-					.uomId(bomUomId)
 					.explodedFromBOMLineId(bomLineId)
 					.build();
 			result.addAll(this.toBuilder()
 					.initialCandidate(lineCandidate)
 					.build()
-					.execute());  // recurse
+					.execute0());  // recurse
 		}
 
 		return result;
 	}
+
+	private List<OrderLineCandidate> reduceCandidates(final List<OrderLineCandidate> result)
+	{
+		final ArrayList<OrderLineCandidate> reduced = new ArrayList<>();
+		for (final OrderLineCandidate candidate : result)
+		{
+			if (candidate.getQty().signum() == 0)
+			{
+				continue;
+			}
+			if (candidate.getQty().signum() < 0)
+			{
+				throw new AdempiereException("Invalid qty: " + candidate.getQty());
+			}
+			if (candidate.getQty().toBigDecimal().compareTo(BigDecimal.ZERO) == 0)
+			{
+				continue;
+			}
+
+			final Optional<OrderLineCandidate> orderLineCandidate = reduced.stream()
+					.filter(cand -> cand.getProductId().equals(candidate.getProductId()))
+					.findFirst();
+			if (orderLineCandidate.isPresent())
+			{
+				final OrderLineCandidate olCandidate = orderLineCandidate.get();
+				if (!olCandidate.getAttributes().equals(candidate.getAttributes()))
+				{
+					throw new AdempiereException("Cannot explode BOM because attributes are different: " + olCandidate.getAttributes() + " and " + candidate.getAttributes());
+				}
+				reduced.remove(olCandidate);
+
+				reduced.add(olCandidate.toBuilder()
+						.qty(olCandidate.getQty()
+								.add(candidate.getQty()))
```

Comment:
I think we shall also handle UOM conversions...

In complex BOMs is pretty common to use different UOMs in different BOMs....

Reply:
replaced with `de.metas.quantity.Quantitys#add`